include/dmlc/parameter.h

```diff
@@ -1012,6 +1013,12 @@ class FieldEntry<float> : public FieldEntryNumeric<FieldEntry<float>, float> {
       throw dmlc::ParamError(os.str());
     }
   }
+
+ protected:
+  // print the value
+  virtual void PrintValue(std::ostream &os, float value) const {  // NOLINT(*)
+    os << std::setprecision(std::numeric_limits<float>::max_digits10) << value;
```

Comment:
This will change the internal state of `os`. Is `os` used somewhere else?

Reply:
Side-effect free alternative:
```suggestion
    char buf[512];
    sprintf(buf, "%.*g", std::numeric_limits<float>::max_digits10, value);
    os << buf;
```

Reply:
`os` is passed by `GetStringValue`, which is a local parameter. It just accept value and return `os.str()`, doesn’t do anything else. So I think `setprecision` is also ok.